### DISCUSSION TITLE
CookieCutter Docker Deployment

### DISCUSSION THREAD

**Discussion by wyattnapier** [Kitware/trame#571] [Q&A]
Hi Jourdain! Thank you for all of your help throughout my project. I'm now working on deployment, so I used the cookiecutter so I had the right boilerplate. I was running into issue when I tried to run `./scripts/build_server.sh`.

I made a new cookiecutter application with just the default core.py and main.py to test if I still got the error when I tried to run `./scripts/build_server.sh`. I did still get the following message: `ERROR: The deploy directory must be mounted into the container at /deploy`. I also tried running Docker Desktop as an administrator and it didn't fix it either. I was wondering if you had any tips to address this.

This is my first time using Docker and I saw in other discussion posts that you mentioned once I understood the cookiecutter and Docker I shouldn't have any problems. I've read through all of the READMEs and related discussion posts. Is there anything in particular that would help me understand them both better?

Thanks again! Hopefully after this issue is resolved I'll be out of your hair :)

**Comment by jourdain** [comment]
On which system do you run it? (Windows?) 
It seems to be related to invalid path provided as arg of the command line within the script.
If you run those command yourself with the proper path, you should be able to go through.

**Reply by wyattnapier** [reply]
Yes, I'm running it on Windows.

This is the `./scripts/build_server.sh` script. 
```
#!/usr/bin/env bash
CURRENT_DIR=`dirname "$0"`

cd $CURRENT_DIR/..
DEPLOY_DIR="$PWD"
cd ../..
ROOT_DIR="$PWD"
echo $ROOT_DIR

docker run --rm          \
    -e TRAME_CLIENT_TYPE=vue3 \
    -v "$DEPLOY_DIR:/deploy" \
    -v "$ROOT_DIR:/local-app"  \
    kitware/trame build
```
I saw that in the previous discussions your recommended getting rid of the local-app line, so I tried that and it didn't work either. I don't think I really understand what the docker run command is trying to do so I'm not sure how to correct the path. Is there any other setup I need to do within the Docker Desktop app besides just opening it? I tried to follow the ReadME, but it feels like I'm missing something.

**Reply by jourdain** [reply]
The script has some logic to get absolute path for the -v args... This is most likely failing on Windows... So just run the unfolded command with absolute path that docker/windows expect.

**Reply by wyattnapier** [reply]
I used the absolute path and I still got the same error: `ERROR: The deploy directory must be mounted into the container at /deploy`. I also tested my docker connection with `docker run hello-world` and it worked properly. Do I have the correct file tree or do I need folders for deploy and server?
![image](https://github.com/user-attachments/assets/7b96aee8-3577-4fc8-8e03-f60e0ac5bebe)

**Reply by jourdain** [reply]
What command line did you run? especially on the `-v xxx:/deploy` part. (That is what the error is telling you about)

**Reply by wyattnapier** [reply]
I tried to run this:
```
docker run --rm          \
    -e TRAME_CLIENT_TYPE=vue3 \
    -v "/c/Users/wnapier/TrameProject/basic-tester/bundles/docker:/deploy" \
    -v "/c/Users/wnapier/TrameProject/basic-tester:/local-app" \
    kitware/trame build
```

**Reply by jourdain** [reply]
Then something is up with docker on windows since you are mounting a directory in /deploy...

**Reply by bourdaisj** [reply]
To add info on top of this, it seems you need to add an extra slash before the source path in the -v option. at least when using git bash it seems to work for me. see https://stackoverflow.com/questions/45971412/docker-volume-format-for-windows

**Comment by clabnet** [comment]
@wyattnapier Same issue.  Is there a strange folder "docker;C" created on root folder. 
I think it is the first command on the bash script not running fine on Windows.  `CURRENT_DIR=`dirname "$0"``
#571 Is there a solution for us poor unfortunate Windows users ?

**Reply by jourdain** [reply]
I guess, on Windows, you should run the command lines yourself for building and running the images. The scripts give you the background on how those command lines are getting built, but like you said, that logic is failing on Windows.
